tests/specs/lemmas.k

```diff
@@ -199,32 +194,37 @@ module LEMMAS-JAVA [symbolic, kast]
     rule _N &Int  0 => 0
     rule  N &Int  N => N
 
-    //orienting symbolic term to be first, converting -Int to +Int for concrete values.
-    rule I +Int B => B          +Int I when #isConcrete(I) andBool notBool #isConcrete(B) [simplification]
-    rule A -Int I => A +Int (0 -Int I) when notBool #isConcrete(A) andBool #isConcrete(I) [simplification]
+    // orienting symbolic term to be first, converting -Int to +Int for concrete values.
+    rule I +Int B => B         +Int I  requires #isConcrete(I) andBool notBool #isConcrete(B) [simplification]
+    rule A -Int I => A +Int (0 -Int I) requires #isConcrete(I) andBool notBool #isConcrete(A) [simplification]
 
-    rule (A +Int I2) +Int I3 => A +Int (I2 +Int I3) when notBool #isConcrete(A) andBool #isConcrete(I2) andBool #isConcrete(I3) [simplification]
+    // associate concrete terms to the left and symbolic terms to the right
+    
+    // // OLD
+    // rule (A  +Int  J) +Int  K  =>  A +Int (J  +Int K) requires #isConcrete(J) andBool #isConcrete(K) andBool notBool #isConcrete(A) [simplification]
+    // rule  I  +Int (B  +Int  K) =>  B +Int (I  +Int K) requires #isConcrete(I) andBool #isConcrete(K) andBool notBool #isConcrete(B) [simplification]
+    // NEW (moves symbolic terms to the fight)
+    rule (A  +Int  J) +Int  K  => (J +Int K) +Int A requires #isConcrete(J) andBool #isConcrete(K) andBool notBool #isConcrete(A) [simplification]
```

Comment:
This rule seems to be moving symbolic terms to the right, but the rule above (line 198) move symbolic terms to the left. This doesn't make sense to me.

Reply:
fixed [here](https://github.com/kframework/evm-semantics/pull/1081/commits/274ff6a354e72a6a27dd0ee3602a992073837fec#r670663973)